### PR TITLE
T44: Introduce TypeScript to existing code

### PR DESCRIPTION
**Pull request by fahadakhan2**
**Description:**
- Installed TypeScript and added tsconfig files in both frontend and backend
- Changed file extensions from js and jsx to ts and tsx
- Added types to the files in the frontend
- Added types to the files in the backend

### CODE REVIEW

**Review by kdata12 — APPROVED** [review]
lgtm!

**Review by kdata12 — COMMENTED** [review]
it seems like you don't have src folder, so we need to fix this tsconfig include array to parse all files in the current directory and subdirectories with the .ts extension

File: backend/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "compilerOptions": {
+    "target": "ES2020",
+    "module": "CommonJS",
+    "strict": true,
+    "esModuleInterop": true,
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true,
+    "outDir": "./dist"
+  },
+  "include": ["src/**/*.ts"]
```
Comment (kdata12) [inline backend/tsconfig.json]:
```suggestion
  "include": ["./**/*.ts"]
```

Comment (fahadakhan2) [reply backend/tsconfig.json]: changed!

**Pull request merged**